### PR TITLE
Switched python package from "nupic" to "htm"

### PR DESCRIPTION
**Pull request by ctrl-z-9000-times**
This changes the python package name from "nupic" to "htm"
This also moves the directories name "nupic" to "htm"
I did not change any of the license notifications.

This is not 100% done yet, but I want to see what CI thinks of it.
I will update this post and add the READY label when it is done.
- [x] TODO: Update htm/optimization code
- [ ] TODO: htm.dummy segfaults

### CODE REVIEW

File: bindings/py/packaging/src/htm/bindings/__init__.py [hunk 1]
```diff
@@ -28,14 +28,13 @@ def _import_helper(name):
...
   # Fast path: see if the module has already been imported.
-  #print("name={}".format(name))
   try:
     return sys.modules[name]
   except KeyError:
     pass
 
   _mod = None
-  basename = name[15:]
+  basename = name[len("htm.bindings."):]
```
Comment (dkeeney) [inline bindings/py/packaging/src/htm/bindings/__init__.py]: Ah, this is something my shell script would not have found 👍

File: bindings/py/packaging/src/htm/bindings/__init__.py [hunk 2]
```diff
@@ -63,15 +62,17 @@ def _import_helper(name):
...
 
-algorithms      = _import_helper('nupic.bindings.algorithms')
-engine_internal = _import_helper('nupic.bindings.engine_internal')
-math            = _import_helper('nupic.bindings.math')
-sdr             = _import_helper('nupic.bindings.sdr')
-encoders        = _import_helper('nupic.bindings.encoders')
+import sys
+if sys.version_info[0] < 3:
+  algorithms      = _import_helper('htm.bindings.algorithms')
+  engine_internal = _import_helper('htm.bindings.engine_internal')
```
Comment (dkeeney) [inline bindings/py/packaging/src/htm/bindings/__init__.py]: > 'htm.bindings.engine_internal'

This is the one I want to change to `htm.bindings.networkapi`

Comment (dkeeney) [reply bindings/py/packaging/src/htm/bindings/__init__.py]: A separate PR perhaps.

File: bindings/py/packaging/src/htm/bindings/__init__.py (hunk 1 above)
Comment (ctrl-z-9000-times) [reply bindings/py/packaging/src/htm/bindings/__init__.py]: CI found this one...

**Comment by ctrl-z-9000-times** [comment]
Thanks for reviewing this, and in a timely manner!

>  Shall we add the rest of the C++ code to this?

I am going to merge this ASAP because this PR blocks any other python code changes.

**Pull request merged**